### PR TITLE
Fix Pet Taming causing seg fault

### PR DESCRIPTION
**Pull request by EmosewaMC**
~~**THIS IS UNTESTED**!!!  Unit tests reveal this change worked however an integration test has not been run quite yet.  If someone would like to prop the build and test pet taming please let me know what is needed to address the issue~~

### CODE REVIEW

**Comment by aronwk-aaron** [comment]
Tested with packed and unpacked clients.
BrickModels is the only special case for not being fully lowercase, so just adding this one exclusion in the asset getter's makes a more centralized and seamless use of the AssetManager

**Review by aronwk-aaron — DISMISSED** [review]

**Review by aronwk-aaron — APPROVED** [review]

**Review by EmosewaMC — COMMENTED** [review]
lgtm

File: dCommon/dClient/AssetManager.cpp [hunk 1]
```diff
@@ -92,17 +94,19 @@ bool AssetManager::GetFile(const char* name, char** data, uint32_t* len) {
...
 	std::replace(fixedName.begin(), fixedName.end(), '\\', '/'); // On the off chance someone has the wrong slashes, force forward slashes
-	auto realPathName = fixedName;
 
-	if (std::filesystem::exists(m_ResPath / realPathName)) {
+	// Special case for unpacked client have BrickModels in upper case
+	if (this->m_AssetBundleType == eAssetBundleType::Unpacked) GeneralUtils::ReplaceInString(fixedName, "brickmodels", "BrickModels");
+
+	if (std::filesystem::exists(m_ResPath / fixedName)) {
 		FILE* file;
 #ifdef _WIN32
```
Comment (Wincent01) [inline dCommon/dClient/AssetManager.cpp:103]: I would prefer to go an abstraction level up to ifsteams here, avoids the ugly preprocessing. Doesn't require changing, just a note.

**Review by Wincent01 — APPROVED** [review]

File: dCommon/dClient/AssetManager.cpp (hunk 1 above)
Comment (Jettford) [reply dCommon/dClient/AssetManager.cpp:103]: I was doing this for speed reasons, no idea how much it actually matters.

**Pull request merged**